### PR TITLE
detect-http-protocol: use inspect/mpm v2 apis

### PR DESCRIPTION
**Pull request by glongo**
Make sure these boxes are signed before submitting your Pull Request -- thank you.

- [x] I have read the contributing guide lines at https://redmine.openinfosecfoundation.org/projects/suricata/wiki/Contributing
- [x] I have signed the Open Information Security Foundation contribution agreement at https://suricata-ids.org/about/contribution-agreement/
- [ ] I have updated the user guide (in doc/userguide/) to reflect the changes made (if applicable)

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
https://redmine.openinfosecfoundation.org/issues/2930

Describe changes:
- This updates inspect/mpm engines apis to v2.

[PRScript](https://redmine.openinfosecfoundation.org/projects/suricata/wiki/PRscript) output (if applicable):
- PR glongo: https://buildbot.openinfosecfoundation.org/builders/glongo/builds/212
- PR glongo-pcap: https://buildbot.openinfosecfoundation.org/builders/glongo-pcap/builds/76

### CODE REVIEW

**Comment by victorjulien** [comment]
@jlucovsky could you review this?

File: src/detect-http-protocol.c [hunk 1]
```diff
@@ -65,166 +65,77 @@
...
-
-    DetectAppLayerInspectEngineRegister(BUFFER_NAME,
-            ALPROTO_HTTP, SIG_FLAG_TOSERVER, HTP_REQUEST_LINE,
-            InspectEngineHttpProtocol);
-    DetectAppLayerInspectEngineRegister(BUFFER_NAME,
-            ALPROTO_HTTP, SIG_FLAG_TOCLIENT, HTP_RESPONSE_LINE,
-            InspectEngineHttpProtocol);
+    sigmatch_table[DETECT_AL_HTTP_PROTOCOL].flags |= SIGMATCH_INFO_STICKY_BUFFER | SIGMATCH_NOOPT;
+
+    DetectAppLayerMpmRegister2(BUFFER_NAME, SIG_FLAG_TOSERVER, 4,
```
Comment (jlucovsky) [inline src/detect-http-protocol.c]: Why is the priority changing from `2` to `4`?

File: src/detect-http-protocol.c [hunk 2]
```diff
@@ -65,166 +65,77 @@
...
+        bstr *str = NULL;
+        htp_tx_t *tx = (htp_tx_t *)txv;
+
+        if (flow_flags & STREAM_TOSERVER)
+            str = tx->request_protocol;
+        else if (flow_flags & STREAM_TOCLIENT)
+            str = tx->response_protocol;
+
+        if (str == NULL)
+            return NULL;
```
Comment (jlucovsky) [inline src/detect-http-protocol.c]: Emit a debug message (`SCLogDebug`) here stating protocol not found.

File: src/detect-http-protocol.c [hunk 3]
```diff
@@ -65,166 +65,77 @@
...
+
+        uint32_t data_len = bstr_size(str);
+        uint8_t *data = bstr_ptr(str);
+        if (data == NULL || data_len == 0)
+            return NULL;
```
Comment (jlucovsky) [inline src/detect-http-protocol.c]: Emit a debug message here stating protocol value not present

File: src/detect-http-protocol.c [hunk 4]
```diff
@@ -65,166 +65,77 @@
 
 #include "util-print.h"
 
-#define KEYWORD_NAME "http_protocol"
+#define KEYWORD_NAME "http.protocol"
```
Comment (jlucovsky) [inline src/detect-http-protocol.c]: Please update https://github.com/OISF/suricata/blob/master/doc/userguide/rules/http-keywords.rst with the new name.

**Comment by victorjulien** [comment]
Could you also add a suricata-verify test?

**Comment by glongo** [comment]
Replaced with https://github.com/OISF/suricata/pull/3814

**Pull request closed without merging**